eidas_node/connector/views.py

```diff
@@ -311,9 +315,20 @@ def create_saml_response(self, issuer: str, audience: Optional[str], destination
         LOGGER.info('[#%r] Created SAML response: id=%r, issuer=%r, in_response_to_id=%r',
                     self.log_id, response.id, response.issuer, response.in_response_to_id)
 
-        # TODO: encrypt the assertion (between response.sign_assertion and response.sign_response)
-        if signature_options and signature_options.get('key_file') and signature_options.get('cert_file'):
+        sign = signature_options and signature_options.get('key_file') and signature_options.get('cert_file')
+        encrypt = encryption_options and encryption_options.get('cert_file')
+
+        if sign:
+            assert signature_options
```

Comment:
These `assertions` seem a little bit redundant since the conditions are right before the code.

Reply:
If you can think of anything better to make mypy happy, I gladly remove the assertions :-)

Reply:
`response.sign_assertion(**cast(Dict[str, Any], signature_options))` does the trick and has no runtime overhead.

Reply:
Done.